### PR TITLE
web: redirect /l/download to App Store

### LINKED ISSUE #666: https://daimo.com/l/download
## Goal
**Create a download link that either takes you to the app or, if not installed, the app store.**

This will go, for example, in the Download button / QR code on the homepage.

- If on iOS, redirect to App Store
- On Android, redirect to Play Store
- Otherwise, show both

### PR DESCRIPTION
**Pull request by dcposch**
fixes #666

### CODE REVIEW

**Comment by dcposch** [comment]
tests failing due to #679

File: apps/daimo-web/src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const url = getAppStoreURL(ua);
+  if (url) return NextResponse.redirect(url);
+  return NextResponse.next();
+}
+
+function getAppStoreURL(ua: string) {
+  const urlAppStore = "https://apps.apple.com/us/app/daimo/id6459700343";
+  const urlPlay = "https://play.google.com/store/apps/details?id=com.daimo";
+  if (ua.includes("Android")) return urlPlay;
+  else if (ua.includes("iPhone")) return urlAppStore;
```
Comment (nalinbhardwaj) [inline apps/daimo-web/src/middleware.ts:22]: looks like we have some utils for this same heuristic in https://github.com/daimo-eth/daimo/blob/master/apps/daimo-web/src/utils/platform.ts, want to dedupe?

Comment (dcposch) [reply apps/daimo-web/src/middleware.ts]: good call, done

File: apps/daimo-web/src/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const url = getAppStoreURL(ua);
+  if (url) return NextResponse.redirect(url);
+  return NextResponse.next();
+}
+
+function getAppStoreURL(ua: string) {
+  const urlAppStore = "https://apps.apple.com/us/app/daimo/id6459700343";
+  const urlPlay = "https://play.google.com/store/apps/details?id=com.daimo";
+  const platform = detectPlatform(ua);
+  if (platform === "ios") return urlAppStore;
```
Comment (nalinbhardwaj) [inline apps/daimo-web/src/middleware.ts:24]: just import and `return downloadMetadata[platform].url` should work no?

**Review by nalinbhardwaj — APPROVED** [review]
lgtm

**Pull request merged**